### ISSUE TITLE
R: set.seed() doesn't work for gblinear

### ISSUE THREAD

**Issue by Eric2333** [dmlc/xgboost#2143]
I'm using version 0.6-4 and I'm trying to make reproducible results with set.seed() right before xgb.cv(). When the booster is gbtree, the result from xgb.cv() is the same every time. But this is not the case for gblinear.

**Comment by Laurae2** [comment]
@Eric2333 If you use multithreaded gblinear, setting a seed is useless because results cannot be reproduced.

Maybe @khotilov has a better explanation for this known issue.

**Comment by hetong007** [comment]
Python also has this problem https://github.com/dmlc/xgboost/issues/2249,  thus it should be a backend issue. Note that with `booster = 'gbtree'` there's no different results, so we can't simply blame multithreading. 

Do you have any idea? @tqchen

**Comment by Laurae2** [comment]
@hetong007 It is multithreading fault for non reproducibility, see here: https://github.com/dmlc/xgboost/issues/2367#issuecomment-305556402

> Another option it to either reduce the number of threads or to run single-threaded. Note that you would also have to do the latter in order to get reproducible results since gblinear employs a "hogwild" (lock-free) parallelization in its cyclical gradient descent updates.

**Comment by hetong007** [comment]
@Laurae2 Ah so it is lock-free. Would be good if there's a switch for it.
